### PR TITLE
[WIP]refactor elections endpoint to enable candidate financial totals export

### PR DESCRIPTION
**Pull request by fec-jli**
## Summary (required)
in order to enable “candidate financial totals” data in election summery page, we need refactor election endpoint to extend ApiResource and overwrite 'build_query()' 

- Resolves # [_Issue number_]
[https://github.com/fecgov/fec-cms/issues/2010](https://github.com/fecgov/fec-cms/issues/2010)
enable the export link for “candidate financial totals” section in election summary page:
example link on prod which is disable now: 
[https://www.fec.gov/data/elections/senate/VA/2018/](https://www.fec.gov/data/elections/senate/VA/2018/)

openFEC Issue: Refactor elections (ElectionView) endpoint to extend ApiResource
[https://github.com/fecgov/openFEC/issues/3254](https://github.com/fecgov/openFEC/issues/3254)

## How to test the changes locally
This PR involves changes on both fec-cms and openFEC side.
1) checkout fec-cms feature branch:  feature/enable_export_link_candidate_info
export FEC_WEB_API_KEY_PUBLIC=xxxxxxx
export FEC_WEB_API_KEY=xxxxxxx
export FEC_CMS_ENVIRONMENT=LOCAL
export FEC_API_URL=http://localhost:5000
export DATABASE_URL=postgresql://:@/cfdm_cms_test

2)check out openFEC feature branch: feature/fix_elections_download
setup openFEC local (env api variables)
export SQLA_CONN=xxxxxxx
export access_key_id=
export secret_access_key=
export bucket=
export region=

3)start redis-server

4)Setup Celery Worker and start
export SQLA_CONN=xxxxxxx
export access_key_id=
export secret_access_key=
export bucket=
export region=

5)example url: [http://127.0.0.1:8000/data/elections/senate/VA/2018/](http://127.0.0.1:8000/data/elections/senate/VA/2018/)
Click "Export" on Candidate financial totals section, download the file.

### CODE REVIEW

**Comment by vrajmohan** [comment]
Apropos of our conversation on a clean `git` history, it would be better to break up the commits into 2:
1. For the refactor to use `ApiResource`
2. Remove the joint fundraisers inflating receipts

Also, it would be easier for the reviewer if you rebased this on `develop` to remove the extraneous commits "d9fb37e6d59a2fe6dc742c8c6442fc49a6c1bba0" and "f920bddf0ddc6299feb6867e0f068e324656f6f4".

**Comment by lbeaufort** [comment]
@vrajmohan since this is targeting the release branch, should @fec-jli rebase off the release branch?

**Comment by vrajmohan** [comment]
Right, good catch!

File: webservices/resources/elections.py [hunk 1]
```diff
@@ -280,6 +288,7 @@ def join_candidate_totals(query, kwargs, totals_model):
         sa.and_(
             CandidateHistory.candidate_id == CandidateCommitteeLink.candidate_id,
             CandidateHistory.two_year_period == CandidateCommitteeLink.fec_election_year,
+            CandidateCommitteeLink.committee_designation.in_(['P', 'A']),
```
Comment (vrajmohan) [inline webservices/resources/elections.py]: A comment on why this is needed may be useful. Also, should we create a test that would add other committee designations and assert that they are *not* picked up?

Comment (jwchumley) [reply webservices/resources/elections.py]: Here is a comment on why this is needed:
Joint Fundraising Committees raise money for multiple candidate committees and transfer that money to those committees. By limiting the committee designations to A and P you eliminate J (joint) and thus do not inflate the candidate's money by including it twice and by including money that was raised and transferred to the other committees in the joint fundraiser.

**Comment by lbeaufort** [comment]
@fec-jli I changed the target branch back to `develop` because this didn't get merged. Is it still a [WIP]?

**Comment by JonellaCulmer** [comment]
@fec-jli Closing this PR in favor of a more up-to-date PR.

**Pull request closed without merging**